src/smart-components/user/user.js

```diff
@@ -0,0 +1,229 @@
+            setExpanded({ ...expanded, [rowData.uuid]: colIndex });
+            // Permissions
+            if (colIndex === 2) {
+                fetchRoleForUser(rowData.uuid);
+            }
+
+            // set all other expanded cells false in this row if we are expanding
+            r[rowIndex].cells.forEach(cell => {
+                if (cell.props) {cell.props.isOpen = false;}
+            });
+            r[rowIndex].cells[colIndex].props.isOpen = true;
+            r[rowIndex].isOpen = true;
+        } else {
+            setExpanded({ ...expanded, [rowData.uuid]: -1 });
+            r[rowIndex].cells[colIndex].props.isOpen = false;
+            r[rowIndex].isOpen = r[rowIndex].cells.some(cell => cell.props && cell.props.isOpen);
+        }
+
+        setRows(r);
+    };
+
+    const onSetPage = (_e, page) => setPage(page);
+    const onPerPageSelect = (_e, perPage) => setPerPage(perPage);
+
+    return (<Stack >
+            <StackItem>
+                <TopToolbar paddingBottm={ false }>
+                    <TopToolbarTitle
+                    title={ username }
+                    description={ `${username}'s roles, groups and permissions.` }
+                    />
+                </TopToolbar>
+            </StackItem>
+            <StackItem>
+                <Section type="content" id={ 'user-detail2' }>
+                <PrimaryToolbar
+                    filterConfig={ { items: [{ label: 'Role',
+                        filterValues: {
+                            value: filter,
+                            placeholder: 'Filter by role name',
+                            onChange: (e, selected) =>setFilter(selected)
+                        }
+                    }]} }
+                    pagination={ { itemCount: roles.meta.count, page, perPage, onSetPage, onPerPageSelect } }
+                />
+                { isLoading
+                ? <ListLoader />
+                : <Table
+                    className='ins-c-activity-table'
+                    aria-label="Collapsible table"
+                    onExpand={ onExpand }
+                    rows={ rows }
+                    cells={ columns }>
+                    <TableHeader >
+
+                    </TableHeader>
+
+                    <TableBody />
+                </Table>
+                }
+                </Section>
+            </StackItem>
+            </Stack>
+    );
+};
+
+User.propTypes = {
+    match: PropTypes.object,
+    fetchRoles: PropTypes.func,
+    fetchRoleForUser: PropTypes.func,
+    roles: PropTypes.array,
+    isLoading: PropTypes.bool,
+    rolesWithAccess: PropTypes.object,
+    isRecordLoading: PropTypes.bool
+};
+
+User.defaultProps = {
+
+};
+
+const mapStateToProps = ({ roleReducer: { roles, isLoading, rolesWithAccess, isRecordLoading }}) => {
+
+    return {
+       roles,
+       isLoading,
+       rolesWithAccess,
+       isRecordLoading
+    };
+};
+
+const mapDispatchToProps = dispatch => {
+    return {
```

Comment:
No need to have return and block, you can directly expose the object, I guess this should use same default pagination as other tables.
```JS
const mapDispatchToProps = dispatch => ({
  fetchRoles: (apiProps = { limit: 10, offset: 0 }) => dispatch(fetchRoles(apiProps)),
  fetchRoleForUser: (uuid) => dispatch(fetchRoleForUser(uuid))
});
```